python-sdk/src/astro/airflow/datasets.py

```diff
@@ -29,8 +29,15 @@ def kwargs_with_datasets(
         and settings.AUTO_ADD_INLETS_OUTLETS
         and input_datasets
     ):
+        # Remove non-dataset objects like Temp tables (unless passed by user)
+        input_datasets = (
+            input_datasets if isinstance(input_datasets, list) else [input_datasets]
+        )
+        input_datasets = [
+            input_ds for input_ds in input_datasets if isinstance(input_ds, Dataset)
+        ]
```

Comment:
hmm 🤔 ? According to the type definition only `Dataset` type is supported anyway.

`input_datasets: list[Dataset] | Dataset | None = None`

Reply:
Yes and No :) --- It is mainly for removing `TempTable` from being added as inlets/outlets... and `TempTable` get generated by the hack in this PR by calling `Table()` without `name`.

Here `Table` is a Dataset class but `TempTable` isn't -- and instead of checking those in each operator, it would be better to check it here.

Reply:
Okay, but then we should update the type definition to reflect that. From your comment it sounds like `TempTable` is supported but will be filtered out from `input_datasets`. So in my opinion it should be: `input_datasets: list[Dataset | TempTable] | Dataset | TempTable | None = None` or?